dev/TEAL.md

```diff
@@ -327,6 +332,10 @@ return stack matches the name of the input value.
 | `extract_uint16` | A uint16 formed from a range of big-endian bytes from A starting at B up to but not including B+2. If B+2 is larger than the array length, the program fails |
 | `extract_uint32` | A uint32 formed from a range of big-endian bytes from A starting at B up to but not including B+4. If B+4 is larger than the array length, the program fails |
 | `extract_uint64` | A uint64 formed from a range of big-endian bytes from A starting at B up to but not including B+8. If B+8 is larger than the array length, the program fails |
+| `replace2 s` | Copy of A with the bytes starting at S replaced by the bytes of B. Fails if S+len(B) exceeds len(A) |
+| `replace3` | Copy of A with the bytes starting at B replaced by the bytes of C. Fails if B+len(C) exceeds len(A) |
+| `base64_decode e` | Decode Base-64 encoded byte-array A. Fail if A is not properly encoded. |
```

Comment:
```suggestion
| `base64_decode e` | Decode Base-64 encoded byte-array A. Fail if A is not properly encoded. WARNING: Usage should be restricted to very rare use cases. |
```

Reply:
I'm very happy with this warning on `json_ref` but I think we can be a bit softer here, since at least b64 is well defined and not all that slow.  How about: "Users of `base64_decode` should consider whether their input could be supplied unencoded to save transaction space and computation within the AVM"

Reply:
Indeed, "Users of base64_decode should consider whether their input could be supplied unencoded to save transaction space and computation within the AVM" looks good to me